src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -703,5 +691,38 @@ public void Assign(string expression)
                 Value = await _engine.DebuggedProcess.MICommandFactory.VarAssign(_internalName, expression);
             });
         }
+
+        #region IDisposable Implementation
+
+        public void Dispose()
+        {
+            this.Dispose(true);
+
+            GC.SuppressFinalize(this);
+        }
```

Comment:
Typical dispose pattern is to throw object disposed after this call for everything except the finalizer. How much extra work would that be?